src/metakb/query.py

```diff
@@ -708,3 +714,66 @@ def _get_therapeutic_agent(in_ta_params: dict) -> TherapeuticAgent:
 
         ta_params["extensions"] = extensions
         return TherapeuticAgent(**ta_params)
+
+    async def batch_search_studies(
+        self,
+        variations: list[str] | None = None,
+    ) -> BatchSearchStudiesService:
+        """Fetch all studies associated with any of the provided variation description
+        strings.
+
+        Because this method could be expanded to include other kinds of search terms,
+        ``variations`` is optionally nullable.
+
+        >>> from metakb.query import QueryHandler
+        >>> qh = QueryHandler()
+        >>> response = await qh.batch_search_studies(["EGFR L858R"])
+        >>> response.study_ids[:3]
+        ['civic.eid:229', 'civic.eid:3811', 'moa.assertion:268']
+
+        All terms are normalized, so redundant terms don't alter search results:
+
+        >>> redundant_response = await qh.batch_search_studies(
+        ...     ["EGFR L858R", "NP_005219.2:p.Leu858Arg"]
+        ... )
+        >>> len(response.study_ids) == len(redundant_response.study_ids)
+        True
+
+        :param variations: a list of variation description strings, e.g.
+            ``["BRAF V600E"]``
+        :return: response object including all matching studies
+        """
+        response = BatchSearchStudiesService(
+            query=BatchSearchStudiesQuery(variations=[]),
+            service_meta_=ServiceMeta(),
+            warnings=[],
+        )
+        if not variations:
+            return response
+
+        for query_variation in variations:
```

Comment:
Since we're calling the variation-normalizer, we probably should only query unique items

```suggestion
        for query_variation in set(variations):
```